src/Functions/h3PointDist.cpp

```diff
@@ -49,7 +49,7 @@ class FunctionH3PointDist final : public IFunction
                 throw Exception(
                         ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
                         "Illegal type {} of argument {} of function {}. Must be Float64",
-                        arg->getName(), i, getName());
+                        arg->getName(), i + 1, getName());
```

Comment:
We could use `FunctionArgumentDescriptor` (src/Functions/FunctionHelpers.cpp) which does all the validation based on an descriptor struct. This would also guarantee consistent error messages (across all functions using `FunctionArgumentDescriptor`).

Reply:
That would indeed be a lot better!

Looks like `validateArgumentType` is currently used in in geohash functions, and I don't see any usage of `validateArgumentsImpl`.
From what I previously saw in the codebase, that could be applied for most functions (there are exceptions like `hascolumnintable` or some functions that accepts 1 or 3 parameters).
So I suppose either it's done from time to time (like in this PR), or a single PR updates the different functions.

I never learnt how to code in cpp, so I prefer doing very small changes. I don't feel comfortable introducing that change in this PR myself.